default.nix

```diff
@@ -131,6 +143,7 @@ let
       shell
       tests
       rust-packages
+      haskell-nix-extra-packages
```

Comment:
Why is this exported via haskell-nix-extra-packages.stackProjectRegenerate when `cabalProjectRegenerate` is exported in the flat name space?

Reply:
`cabalProjectRegenerate` does not make use of `haskell.nix`/ `nix-tools`, hence it is ok to use directly (not through overlay).
Conversely, since `stackProjectRegenerate` make use of `haskell.nix`/ `nix-tools` it should be used in projects through the `haskell-nix-extra` overlay in order for the script to use the same version of `haskell.nix`/ `nix-tools` that the project is using. 

`haskell-nix-extra-packages` is mostly there for the script to be built on hydra.